### PR TITLE
INDY-2316: update test_recover_taa_from_ledger

### PR DESCRIPTION
**Pull request by Toktar**
Signed-off-by: toktar <renata.toktar@dsr-corporation.com>

### CODE REVIEW

File: indy_node/test/txn_author_agreement/test_recover_taa_from_ledger.py [hunk 1]
```diff
@@ -113,4 +119,16 @@ def test_recover_taa_from_ledger(txnPoolNodeSet,
...
     assert res_1['result']['data'][TXN_AUTHOR_AGREEMENT_VERSION] == version_1
     assert res_1['result']['data'][TXN_AUTHOR_AGREEMENT_RETIREMENT_TS] == retired_time
-    assert res_1['result']['data'][TXN_AUTHOR_AGREEMENT_RATIFICATION_TS] == get_txn_time(res_0['result'])
+    assert res_1['result']['data'][TXN_AUTHOR_AGREEMENT_RATIFICATION_TS] == taa_1_ratification_ts
+
+    # Step 14. Disable TAA written using old handler, make sure ratification date is not spoiled
+    disable_res = sdk_send_txn_author_agreement_disable(looper, sdk_pool_handle, sdk_wallet_trustee)[1]
+    res_0 = sdk_get_txn_author_agreement(looper, sdk_pool_handle, sdk_wallet_trustee, version=version_0)[1]
+
+    assert TXN_AUTHOR_AGREEMENT_DIGEST in res_0['result']['data']
```
Comment (ashcherbakov) [inline indy_node/test/txn_author_agreement/test_recover_taa_from_ledger.py]: Why don't we check this for the old TAA with `version_1` as well?

Comment (Toktar) [reply indy_node/test/txn_author_agreement/test_recover_taa_from_ledger.py]: Added a new check

File: indy_node/test/txn_author_agreement/test_recover_taa_from_ledger.py [hunk 2]
```diff
@@ -106,11 +106,24 @@ def test_recover_taa_from_ledger(txnPoolNodeSet,
...
+    check_new_format_taa(res_1, version_1, taa_1_ratification_ts, retired_time_in_past)
+
+
+def check_new_format_taa(result, version, ratification_ts, retired_time=None):
+    assert TXN_AUTHOR_AGREEMENT_DIGEST in result['result']['data']
+    assert TXN_AUTHOR_AGREEMENT_RATIFICATION_TS in result['result']['data']
+    assert result['result']['data'][TXN_AUTHOR_AGREEMENT_VERSION] == version
+    assert result['result']['data'][TXN_AUTHOR_AGREEMENT_RATIFICATION_TS] == ratification_ts
+    if retired_time:
+        assert result['result']['data'][TXN_AUTHOR_AGREEMENT_RETIREMENT_TS] == retired_time
```
Comment (skhoroshavin) [inline indy_node/test/txn_author_agreement/test_recover_taa_from_ledger.py]: Should we also add else clause with `assert TXN_AUTHOR_AGREEMENT_RETIREMENT_TS not in result['result']['data']`?

File: indy_node/test/txn_author_agreement/test_recover_taa_from_ledger.py [hunk 3]
```diff
@@ -106,11 +106,24 @@ def test_recover_taa_from_ledger(txnPoolNodeSet,
...
+    disable_res = sdk_send_txn_author_agreement_disable(looper, sdk_pool_handle, sdk_wallet_trustee)[1]
+
+    res_0 = sdk_get_txn_author_agreement(looper, sdk_pool_handle, sdk_wallet_trustee, version=version_0)[1]
+    check_new_format_taa(res_0, version_0, taa_0_ratification_ts, get_txn_time(disable_res['result']))
+
+    res_1 = sdk_get_txn_author_agreement(looper, sdk_pool_handle, sdk_wallet_trustee, version=version_1)[1]
+    check_new_format_taa(res_1, version_1, taa_1_ratification_ts, retired_time_in_past)
+
+
+def check_new_format_taa(result, version, ratification_ts, retired_time=None):
```
Comment (skhoroshavin) [inline indy_node/test/txn_author_agreement/test_recover_taa_from_ledger.py]: If we create helper function for checking something I'd prefer have more clear name, like `check_result_contains_expected_taa_data`. Another possibility is to create function like this:
```
def expected_taa_data(version, ratification_ts, ...):
   result = {
      TXN_AUTHOR_AGREEMENT_VERSION: version,
      ...
   }
   ...
   return result
```
and then use it like this:
```
assert result['result']['data'] == expected_taa_data(version, ratification_ts, ...)
```

**Comment by Toktar** [comment]
test this please

**Pull request merged**